File: YemenSchoolsV1.Domain/Entities/Term.cs
namespace YemenSchoolsV1.Domain.Entities
{
    public class Term
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
        public Guid AcademicYearId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        // Navigation Property
        public AcademicYear AcademicYear { get; set; } = null!;
        public ICollection<SectionSubject> SectionSubjects { get; set; } = [];

    }
}

File: YemenSchoolsV1.Clint/src/app/features/school-dashboard/term/models/term.ts
export interface Term {
    id: string;
    name: string;
    academicYearName: string;
    startDate: Date;
    endtDate: Date;
}

export interface TermDto extends Term { }

export interface CreateTermDto {
    name: string;
    startDate: Date;
    endDate: Date;
    academicYearId: string;
}

export interface UpdateTermDto extends CreateTermDto {
    id: string;
}
